### PR TITLE
Pipeline

### PR DESCRIPTION
**Pull request by ben-heil**
RE "are you going to make a snakemake pipeline from these shell scripts at some point?" Yes, and that point is now.

~This PR is currently a draft. I'll finalize it after running it overnight to make sure everything is working correctly~ should be ready now!

Changes:
- Created a Snakefile to run all the steps of the pipeline
- `split_pairwise_networks.py` -> `split_pairwise_network.py`; I modified the script to take one pair at a time to allow for paralellization. Git was very confused by this since I also made some changes to the script itself, and claimed I created an entirely new file
- `build_pairwise_networks.py` -> `indices/build_pairwise_network.py`; Git was fine with this one :shrug: 
- Update `download_citations.sh` to point to the correct directory (or at least what the correct directory will be. Since I have multiple projects on my computer right now the citation data is currently stored on a bigger, slower, spinning disk
- Added snakemake to the conda env

### CODE REVIEW

File: Snakefile [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+wildcard_constraints:
+    # Random seeds should be numbers
+    shuffle="\d+",
+    # The headings wildcard used in shuffle_networks can contain letters, underscores,
+    # and dashes, but no other characters such as numbers
+    heading="[a-z_-]+",
+    heading1="[a-z_-]+",
+    heading2="[a-z_-]+",
+
+ruleorder: split_combined_shuffled_networks > shuffle_networks
```
Comment (jjc2718) [inline Snakefile:24]: Why do you need `ruleorder` here? I thought in Snakemake the order rules run in was dictated by the input and output files that need to be generated - does it not automatically handle this case?

File: indices/split_pairwise_network.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import networkx as nx
+
+from utils import parse_mesh_headings
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('in_file',
+                        help='The file to be split',
+                        default='data/networks')
```
Comment (jjc2718) [inline indices/split_pairwise_network.py]: This default isn't a file, though, is it? It looks like a directory.

I know you're providing this argument when you call this script in `snakemake` so it probably doesn't matter too much.

**Review by jjc2718 — APPROVED** [review]
Looks good!

File: Snakefile (hunk 1 above)
Comment (ben-heil) [reply Snakefile:24]: As far as I can tell, Snakemake looks at a target file from `rule all` and says "which rule can create that output?" For whatever reason it is convinced that `shuffle_networks` can create files from `split_combined_shuffled_networks`, despite the fact that their outputs are in different directories. I think it tries to coerce the `heading` wildcard to eat some of the directory portion of the output file, despite the fact that slashes are explicitly disallowed :shrug:

In general, the AmbiguousRuleExceptions are weird and somewhat broken and I try to just find what fixes them and not think too hard about the underlying software https://github.com/snakemake/snakemake/issues/674

File: indices/split_pairwise_network.py (hunk 2 above)
Comment (ben-heil) [reply indices/split_pairwise_network.py]: Good catch! This script used to take in a directory and I updated the argument name but didn't remove the default

**Pull request merged**